LykkeApi2/Controllers/AssetsController.cs

```diff
@@ -228,18 +228,28 @@ private static AssetDescriptionModel ConvertToAssetDescription(AssetExtendedInfo
 
         [Authorize]
         [HttpPost("baseAsset")]
-        [ProducesResponseType((int)HttpStatusCode.OK)]
-        [ProducesResponseType((int)HttpStatusCode.BadRequest)]
+        [ProducesResponseType((int) HttpStatusCode.OK)]
+        [ProducesResponseType((int) HttpStatusCode.BadRequest)]
+        [ProducesResponseType((int) HttpStatusCode.NotFound)]
         public async Task<IActionResult> SetBaseAsset([FromBody] BaseAssetUpdateModel model)
         {
             try
             {
+                var assetResponse = await _assetsService.AssetGetWithHttpMessagesAsync(model.BaseAsssetId);
+                var asset = assetResponse?.Body as Asset;
+
+                if (asset == null)
+                    return NotFound();
+
+                if (!asset.IsBase)
+                    return BadRequest(new {message = "Asset can't be set as base"});
+
                 await _clientAccountSettingsClient.SetBaseAssetAsync(_requestContext.ClientId, model.BaseAsssetId);
             }
             catch (Exception e)
             {
                 await _log.WriteFatalErrorAsync(nameof(AssetsController), nameof(SetBaseAsset), e);
-                return BadRequest(new { message = e.Message });
+                return BadRequest(new {message = e.Message});

```

Comment:
I think we should not return exception details to client. It is an InternalServerError.